src/backend/distributed/master/master_create_shards.c

```diff
@@ -386,6 +409,376 @@ CreateReferenceTableShard(Oid distributedTableId)
 }
 
 
+/*
+ * CreateCitusLocalTableShard creates the one and only shard of the citus
+ * local table lazily. That means, this function suffixes shardId to:
+ *  - relation name,
+ *  - all the objects "defined on" the relation and
+ *  - the foreign keys referencing to the relation.
+ */
+void
+CreateCitusLocalTableShard(Oid relationOid, uint64 shardId)
```

Comment:
same here: would be nice to move this logic to a separate file. I think a single new file would be nice.